marklogic-data-hub-central/ui/src/components/advanced-settings/advanced-settings-dialog.test.tsx

```diff
@@ -128,83 +169,147 @@ describe('Update data load settings component', () => {
+    fireEvent.change(getByLabelText('processors-textarea'), { target: { value: '{"badJSON": "noClosingBracket"' }});
+    expect(queryAllByText('Invalid JSON').length === 3);
+
+    // Valid JSON
+    fireEvent.change(getByLabelText('headers-textarea'), { target: { value: '{"goodJSON": true}' }});
+    expect(queryAllByText('Invalid JSON').length === 2);
+    fireEvent.change(getByLabelText('processors-textarea'), { target: { value: '{"goodJSON": true}' }});
+    expect(queryAllByText('Invalid JSON').length === 1);
+    fireEvent.change(getByLabelText('processors-textarea'), { target: { value: '{"goodJSON": true}' }});
+    expect(queryAllByText('Invalid JSON').length === 0);
+
   });
 
-  test('Verify read only users cannot edit settings', () => {
-    const { getByText, getByPlaceholderText, getByRole } = render(<AdvancedSettingsDialog {...data.advancedSettings} canWrite={false} />);
+  test('Verify read only users cannot edit settings', async () => {
+    let getByText, getByPlaceholderText;
+    await act(async () => {
+      const renderResults = render(
+        <AdvancedSettingsDialog {...data.advancedMapping} canWrite={false} />
+      );
+      getByText = renderResults.getByText;
+      getByPlaceholderText = renderResults.getByPlaceholderText;
+    });
+
     expect(document.querySelector('#sourceDatabase')).toHaveClass('ant-select-disabled');
     expect(document.querySelector('#targetDatabase')).toHaveClass('ant-select-disabled');
     expect(document.querySelector('#additionalColl')).toHaveClass('ant-select-disabled');
     expect(getByPlaceholderText('Please enter target permissions')).toBeDisabled();
     expect(getByPlaceholderText('Please enter batch size')).toBeDisabled();
+    expect(document.querySelector('#headers')).toHaveClass('ant-input-disabled');
+    expect(document.querySelector('#targetFormat')).toHaveClass('ant-select-disabled');
     expect(document.querySelector('#provGranularity')).toHaveClass('ant-select-disabled');
 
+    fireEvent.click(getByText('Processors'));
+    expect(document.querySelector('#processors')).toHaveClass('ant-input-disabled');
+
     fireEvent.click(getByText('Custom Hook'));
-    expect(getByPlaceholderText('Please enter module')).toBeDisabled();
-    expect(getByPlaceholderText('Please enter parameters')).toBeDisabled();
-    expect(getByPlaceholderText('Please enter user information')).toBeDisabled();
-    expect(getByRole('switch')).toBeDisabled();
+    expect(document.querySelector('#customHook')).toHaveClass('ant-input-disabled');
   });
 
-  test('Verify put is called when Mapping configuration is saved', async () => {
-    //Enhance this test once DHFPROD-4712 is fixed
-    axiosMock.put.mockImplementationOnce(jest.fn(() => Promise.resolve({ status: 200, data: {} })));
-    const { getByText } = render(<AdvancedSettingsDialog {...data.advancedSettings} />);
+  test('Verify post is called when Mapping settings are saved', async () => {
```

Comment:
I switched to POST from PUT here because PUT no longer worked for updating the new JSON fields. We should discuss with @ryanjdew and update Advanced Settings (and possibly other) CRUD across app if necessary so we aren't testing verbs unnecessarily.